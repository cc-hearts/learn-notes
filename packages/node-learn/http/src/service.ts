import Server from './http.js'
import RouterTest from './router-test.js'

const app = new Server()


app.use('/api', RouterTest)



app.listen(9000)