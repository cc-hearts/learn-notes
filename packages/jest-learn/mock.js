import { getName } from "./service";
export function searchName(name) {
  const names = getName();
  console.log(names)
  return names.filter((val) => val === name);
}
