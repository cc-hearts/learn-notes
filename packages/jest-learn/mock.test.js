import { searchName } from "./mock";
import { getName } from "./service";
// 替换service模块 这样替换的结果只能是全局唯一的 使用mockImplementation 达到每个测试用例都有自己的mock数据
// jest.mock("./service", () => ({
//   getName: jest.fn(() => ["key", "join"]),
// }));

// 用 fn的callback 代替 数据
jest.mock("./service", () => ({
  getName: jest.fn(),
}));
describe("mock searchName result", () => {
  it("should return target result when found search", () => {
    const keyword = "key";
    getName.mockImplementation(() => ["key", "join"]);
    const result = searchName(keyword);
    expect(result).toEqual([keyword]);
  });

  it("should return empty when not found search", () => {
    const keyword = "key";
    getName.mockImplementation(() => ["join", "split"]);
    const result = searchName(keyword);
    expect(result).toEqual([]);
  });

  it("should generate a new snapshot", () => {
    const key = "join";
    getName.mockImplementation(() => ["join", "split"]);
    const result = searchName(key);
    expect(result).toMatchSnapshot(`[join]`);
  });

  // yarn jest -u 会更新行内快照的值 如果需要重新生成行内的快照 可以再次使用 -u
  it("should generate a new InlineSnapshot", () => {
    const key = "key";
    getName.mockImplementation(() => ["join", "split"]);
    const result = searchName(key);
    expect(result).toMatchInlineSnapshot(`[]`);
  });
});
