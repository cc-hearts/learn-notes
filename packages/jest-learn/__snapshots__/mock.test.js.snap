// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock searchName result should generate a new snapshot: [join] 1`] = `
[
  "join",
]
`;
